models/validators/UniqueValidator.cfc

```diff
@@ -2,64 +2,113 @@
  * Copyright since 2020 by Ortus Solutions, Corp
  * www.ortussolutions.com
  * ---
- * This validator checks the database according to validation data for field uniqueness
- * - table : The table name to seek
- * - column : The column to evaluate for uniqueness or defaults to the name of the field
+ * This validator validates if a value is unique in a database
  */
-component
-	extends  ="BaseValidator"
-	accessors="true"
-	singleton
-{
+component accessors="true" singleton {
+
+	property name="name";
+	property name="res" 				inject="model:resourceService@cbi18n";
+    property name="moduleService" inject="coldbox:moduleService";
+    property name="wirebox" inject="wirebox";
 
 	/**
 	 * Constructor
 	 */
-	UniqueValidator function init(){
-		variables.name = "Unique";
+	QUniqueValidator function init() {
+		variables.name = "QUniqueValidator";
 		return this;
 	}
 
 	/**
 	 * Will check if an incoming value validates
-	 *
 	 * @validationResult The result object of the validation
 	 * @target The target object to validate on
 	 * @field The field on the target object to validate on
 	 * @targetValue The target value to validate
 	 * @validationData The validation data the validator was created with
 	 */
 	boolean function validate(
-		required any validationResult,
+		required  validationResult,
 		required any target,
 		required string field,
 		any targetValue,
 		any validationData
-	){
-		// Default the target column
-		var targetColumn = ( isNull( arguments.validationData.column ) ? arguments.field : arguments.validationData.column );
-		// Query it
-		var exists       = queryExecute(
-			"SELECT 1 FROM #arguments.validationData.table# WHERE #targetColumn# = ?",
-			[ arguments.targetValue ]
-		).recordCount > 0;
+	) {
 
-		if ( !exists ) {
-			return true;
-		}
+        //check to see if cborm is loaded and if this is a cform entity
+        if ( moduleService.isModuleActive("cborm") and getMetadata(arguments.target).persistent ?: false ) {
```

Comment:
let's not worry about cborm. cborm has it's own validator.

Reply:
Ok, removed